src/ILLink.CodeFix/RequiresUnreferencedCodeCodeFixProvider.cs

```diff
@@ -5,120 +5,37 @@
-using Microsoft.CodeAnalysis.Simplification;
 
 namespace ILLink.CodeFix
 {
 	[ExportCodeFixProvider (LanguageNames.CSharp, Name = nameof (RequiresUnreferencedCodeCodeFixProvider)), Shared]
-	public class RequiresUnreferencedCodeCodeFixProvider : CodeFixProvider
+	public class RequiresUnreferencedCodeCodeFixProvider : BaseAttributeCodeFixProvider
 	{
 		private const string s_title = "Add RequiresUnreferencedCode attribute to parent method";
 
 		public sealed override ImmutableArray<string> FixableDiagnosticIds
 			=> ImmutableArray.Create (RequiresUnreferencedCodeAnalyzer.DiagnosticId);
 
-		public sealed override FixAllProvider GetFixAllProvider ()
-		{
-			// See https://github.com/dotnet/roslyn/blob/master/docs/analyzers/FixAllProvider.md for more information on Fix All Providers
-			return WellKnownFixAllProviders.BatchFixer;
-		}
-
 		public sealed override async Task RegisterCodeFixesAsync (CodeFixContext context)
 		{
-			var root = await context.Document.GetSyntaxRootAsync (context.CancellationToken).ConfigureAwait (false);
-
-			var diagnostic = context.Diagnostics.First ();
-			var diagnosticSpan = diagnostic.Location.SourceSpan;
-
-			// Find the containing method
-			SyntaxNode targetNode = root!.FindNode (diagnosticSpan);
-			CSharpSyntaxNode? declarationSyntax = null;
-			for (SyntaxNode? current = targetNode.Parent;
-				 current is not null;
-				 current = current.Parent) {
-				if (current is LambdaExpressionSyntax) {
-					return;
-				} else if (current.IsKind (SyntaxKind.LocalFunctionStatement)
-					  || current is BaseMethodDeclarationSyntax) {
-					declarationSyntax = (CSharpSyntaxNode) current;
-					break;
-				}
-			}
-
-			if (declarationSyntax is not null) {
-				var semanticModel = await context.Document
-					.GetSemanticModelAsync (context.CancellationToken).ConfigureAwait (false);
-				var symbol = semanticModel!.Compilation.GetTypeByMetadataName (
-					RequiresUnreferencedCodeAnalyzer.FullyQualifiedRequiresUnreferencedCodeAttribute);
-
-				// Register a code action that will invoke the fix.
-				context.RegisterCodeFix (
-					CodeAction.Create (
-						title: s_title,
-						createChangedDocument: c => AddRequiresUnreferencedCode (
-							context.Document, root, targetNode, declarationSyntax, symbol!, c),
-						equivalenceKey: s_title),
-					diagnostic);
-
-			}
+			await BaseRegisterCodeFixesAsync (context, AttributeableParentTargets.Method, RequiresUnreferencedCodeAnalyzer.FullyQualifiedRequiresUnreferencedCodeAttribute, s_title);
 		}
 
-		private static async Task<Document> AddRequiresUnreferencedCode (
-			Document document,
-			SyntaxNode root,
-			SyntaxNode targetNode,
-			CSharpSyntaxNode containingDecl,
-			ITypeSymbol requiresUnreferencedCodeSymbol,
-			CancellationToken cancellationToken)
+		internal override SyntaxNode[] SetAttributeArguments (SemanticModel semanticModel, SyntaxNode targetNode, CSharpSyntaxNode containingDecl, SyntaxGenerator generator, Diagnostic diagnostic)
```

Comment:
`Set` methods set particular state. I think you want `Get` since you're returning the result.